### PR TITLE
WIP: phone-field

### PR DESCRIPTION
**Pull request by ovkulkarni**
Just tracking right now, do not merge until finished. I will comment when I am done working on it.

### CODE REVIEW

File: intranet/apps/preferences/tests.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8 -*-
+from django.core.urlresolvers import reverse
+from ...test.ion_test import IonTestCase
+from mock import patch
+import views
```
Comment (tjhsst-ion) [inline intranet/apps/preferences/tests.py]: Codacy Issue found: [Unused import views](https://www.codacy.com/app/tjcsl/ion/file/2378508907/issues/source?bid=3176164&fileBranchId=3225173#l5)

File: intranet/apps/preferences/tests.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        self.login()
+        response = self.client.post(reverse('preferences'), {'showpictures-self': ['on'], 'mobile_phone': ['5555555555'], 'showschedule-self': ['on'], 'home_phone': ['5555555555'], 'webpage_0': ['https://ion.tjhsst.edu/logout'], 'receive_push_notifications': ['on'], 'email_0': ['awilliam@tjhsst.edu'], 'showbirthday-self': ['on'], 'preferred_photo': ['AUTO'], 'other_phone_0': ['5555555555']})
+        self.assertEqual(response.status_code, 302)
+        response = self.client.post(reverse('preferences'), "{}" )
+        self.assertEqual(response.status_code, 404)
+
+    @patch('views.logger')
+    def test_clear_preferences(self, mock_logger):
+        self.login()
+        response = self.client.post(reverse('preferences'), "{'showpictures-self': ['on'], 'mobile_phone': [''], 'showschedule-self': ['on'], 'home_phone': [''], 'webpage_0': ['https://ion.tjhsst.edu/logout'], 'receive_push_notifications': ['on'], 'email_0': ['awilliam@tjhsst.edu'], 'showbirthday-self': ['on'], 'preferred_photo': ['AUTO'], 'other_phone_0': ['']}" )
```
Comment (tjhsst-ion) [inline intranet/apps/preferences/tests.py]: Codacy Issue found: [Unused variable 'response'](https://www.codacy.com/app/tjcsl/ion/file/2378508907/issues/source?bid=3176164&fileBranchId=3225173#l23)

File: intranet/apps/preferences/tests.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+from ...test.ion_test import IonTestCase
+from mock import patch
+import views
+
+class PreferencesTest(IonTestCase):
+    def test_get_preferences(self):
+        self.login()
+        response = self.client.get(reverse('preferences'))
+        self.assertEqual(response.status_code, 200)
+
```
Comment (tjhsst-ion) [inline intranet/apps/preferences/tests.py]: Codacy Issue found: [Trailing whitespace](https://www.codacy.com/app/tjcsl/ion/file/2378508907/issues/source?bid=3176164&fileBranchId=3225173#l12)

File: intranet/apps/preferences/tests.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
+from django.core.urlresolvers import reverse
+from ...test.ion_test import IonTestCase
+from mock import patch
+from . import views
```
Comment (tjhsst-ion) [inline intranet/apps/preferences/tests.py]: Codacy Issue found: [Unused import views](https://www.codacy.com/app/tjcsl/ion/file/2378664538/issues/source?bid=3176164&fileBranchId=3225173#l5)

File: intranet/apps/preferences/tests.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+
+class PreferencesTest(IonTestCase):
+
+    def test_get_preferences(self):
+        self.login()
+        response = self.client.get(reverse('preferences'))
```
Comment (tjhsst-ion) [inline intranet/apps/preferences/tests.py]: Codacy Issue found: [Trailing whitespace](https://www.codacy.com/app/tjcsl/ion/file/2378664538/issues/source?bid=3176164&fileBranchId=3225173#l12)

File: intranet/apps/preferences/tests.py [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+    def test_set_preferences(self):
+        self.login()
+        response = self.client.post(reverse('preferences'),
+                                    {'showpictures-self': ['on'],
+                                     'mobile_phone': ['5555555555'],
+                                     'showschedule-self': ['on'],
+                                     'home_phone': ['5555555555'],
+                                     'webpage_0': ['https://ion.tjhsst.edu/logout'],
+                                     'receive_push_notifications': ['on'],
```
Comment (tjhsst-ion) [inline intranet/apps/preferences/tests.py]: Codacy Issue found: [Unused variable 'response'](https://www.codacy.com/app/tjcsl/ion/file/2378664538/issues/source?bid=3176164&fileBranchId=3225173#l23)

File: intranet/apps/preferences/tests.py [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+                                     'email_0': ['awilliam@tjhsst.edu'],
+                                     'showbirthday-self': ['on'],
+                                     'preferred_photo': ['AUTO'],
+                                     'other_phone_0': ['5555555555']})
+        self.assertEqual(response.status_code, 302)
+        response = self.client.post(reverse('preferences'), "{}")
+        self.assertEqual(response.status_code, 404)
+
+    @patch('views.logger')
```
Comment (pefoley2) [inline intranet/apps/preferences/tests.py]: Don't mock out the logger, use https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertLogs

**Comment by pefoley2** [comment]
Looks fairly good, but please resolve the outstanding comments and make sure ./manage.py test passes.
Might as well squash the commits while you're at it.

File: intranet/apps/preferences/tests.py [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+                                     'showbirthday-self': ['on'],
+                                     'preferred_photo': ['AUTO'],
+                                     'other_phone_0': ['5555555555']})
+        self.assertEqual(response.status_code, 302)
+        response = self.client.post(reverse('preferences'), "{}")
+        self.assertEqual(response.status_code, 404)
+
+    def test_clear_preferences(self):
+        self.login()
+        response = self.client.post(reverse('preferences'),
```
Comment (tjhsst-ion) [inline intranet/apps/preferences/tests.py]: Codacy Issue found: [Unused variable 'response'](https://www.codacy.com/app/tjcsl/ion/file/2379159308/issues/source?bid=3176164&fileBranchId=3225173#l32)

Comment (pefoley2) [inline intranet/apps/preferences/tests.py]: Make sure to assert on response.status_code here as well.

File: intranet/apps/preferences/tests.py [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+                                     'preferred_photo': ['AUTO'],
+                                     'other_phone_0': ['5555555555']})
+        self.assertEqual(response.status_code, 302)
+        response = self.client.post(reverse('preferences'), "{}")
+        self.assertEqual(response.status_code, 404)
+
+    def test_clear_preferences(self):
+        self.login()
+        with self.assertLogs("intranet.apps.preferences.views", level="DEBUG") as logger:
+            response = self.client.post(reverse('preferences'),
```
Comment (tjhsst-ion) [inline intranet/apps/preferences/tests.py]: Codacy Issue found: [Unused variable 'response'](https://www.codacy.com/app/tjcsl/ion/file/2379227851/issues/source?bid=3176164&fileBranchId=3225173#l32)

File: intranet/apps/preferences/tests.py [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+                                     'preferred_photo': ['AUTO'],
+                                     'other_phone_0': ['5555555555']})
+        self.assertEqual(response.status_code, 200)
+        response = self.client.post(reverse('preferences'), {})
+        self.assertEqual(response.status_code, 404)
+
+    def test_clear_preferences(self):
+        self.login()
+        with self.assertLogs("intranet.apps.preferences.views", level="DEBUG") as logger:
+            response = self.client.post(reverse('preferences'),
```
Comment (tjhsst-ion) [inline intranet/apps/preferences/tests.py]: Codacy Issue found: [Unused variable 'response'](https://www.codacy.com/app/tjcsl/ion/file/2379315884/issues/source?bid=3176164&fileBranchId=3225173#l32)

**Pull request closed without merging**